src/pmpo_materialPoints.cpp

```diff
@@ -68,6 +68,18 @@ MaterialPoints::~MaterialPoints() {
     delete MPs;
 }
 
+void MaterialPoints::rebuild(IntView addedMP2elm, IntView addedMPAppID) {
+  auto addedSlices = createInternalMemberViews(addedMP2elm.size(), addedMP2elm, addedMPAppID);
+  IntView curr_elem("curr_elem", MPs->capacity());
+  auto setElem = PS_LAMBDA(const int& e, const int& mp, const int& mask){
+    if(mask) curr_elem(mp) = e;
+    else curr_elem(mp) = -1;
+  };
+  parallel_for(setElem, "setElem");
+
+  MPs->rebuild(curr_elem, addedMP2elm, addedSlices);
```

Comment:
This is fine. I would have assumed PUMIPic has a 'rebuild' method without taking 'curr_elem'. Again, thats ok. Note that for a real case, MPs will migrate in (i.e., added to this process/MPI rank) and also migrate out (deleted from this process/MPI rank),  and in which case we will use updated 'curr_elem'